### PR TITLE
New: Renamed Blacklist to Blocklist

### LINKED ISSUE #1182: Pull Sonarr commit 'New: Renamed Blacklist to Blocklist'
Pull https://github.com/Sonarr/Sonarr/commit/ead1371846b1f19cd49928052be0128bf7ccd41f

```
git cherry-pick -ex ead1371846b1f19cd49928052be0128bf7ccd41f
```

This commit has 45 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

[AB#1400](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1400)

### PR DESCRIPTION
**Pull request by RetroDadson**
(cherry picked from commit ead1371846b1f19cd49928052be0128bf7ccd41f)

#### Database Migration
YES - 014

#### Description
As part of the move to a more inclusive society, words like whitelist/blacklist are being moved away from in favour of the likes of allow list and blocklist. This replaces any references to the phrase blacklist with blocklist in the following areas: 
- Frontend User Visible
- Frontend Developer Visible
- Backend User Visible (e.g. logs) 
- Backend Developer Visible (with the exception of previous database migrations) 
- SQLLite Database

Blocklist was chosen in part via the Google Style Guide which discusses this matter and in tandem with both the Sonarr and Radarr/Lidarr/Readarr teams within the Sonarr Discord collaboration channel (https://discord.com/channels/383686866005917708/542785748768915456/859534462839685160)

#### Todos
- [x] Tests
- [x] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [ ] [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #1182

### CODE REVIEW

File: src/Readarr.Api.V1/Blacklist/BlacklistController.cs [hunk 1]
```diff
@@ -1,7 +1,9 @@
+// Blacklist has been deprecated for blocklist.
```
Comment (RetroDadson) [inline src/Readarr.Api.V1/Blacklist/BlacklistController.cs]: Retained for backwards compatibility same as Sonarr

**Comment by RetroDadson** [comment]
Once Radarr/Radarr#6558 has landed I will move the frontend migration to this as well

**Comment by RetroDadson** [comment]
@Qstick  should be good to go on this one now as well

**Review by Qstick — APPROVED** [review]

**Comment by Kyr4l** [comment]
woke trash

**Comment by bakerboy448** [comment]
@Kyr4l just a warning: unproductive, pointless comments like yours are unnecessary and are considered spam.

**Pull request merged**